File: bank-service/src/main/java/rs/edu/raf/repository/CreditRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.credit.Credit;

import java.math.BigDecimal;
import java.util.List;

@Repository
public interface CreditRepository extends JpaRepository<Credit, Long> {
    List<Credit> findAllByRemainingDebtGreaterThan(BigDecimal balance);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/CreditRepository.kt
package org.example.bankservicekotlin.repository

import org.example.bankservicekotlin.model.credit.Credit
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.math.BigDecimal

@Repository
interface CreditRepository: JpaRepository<Credit, Long> {
    fun findAllByRemainingDebtGreaterThan(balance: BigDecimal): List<Credit>
}